api/core/storage/mongo/partials/task_variants.py

```diff
@@ -43,3 +43,9 @@ async def get_latest_task_variant(
             return TaskVariantDocument.model_validate(doc).to_resource()
 
         return None
+
+    async def get_task_variant_by_uid(self, task_uid: int) -> SerializableTaskVariant:
+        doc = await self._find_one(filter={"task_uid": task_uid})
```

Comment:
maybe add an order to make sure it's deterministic ?

Reply:
https://github.com/WorkflowAI/WorkflowAI/pull/247/commits/20a2b9fbf31dd0a7f9ac6dd6e9436bb14a789ddd